contracts/ServiceManagerToken.sol

```diff
@@ -98,6 +98,11 @@ contract ServiceManagerToken is GenericManager {
             uint256 numAgents = agentParams.length;
             uint256[] memory bonds = new uint256[](numAgents);
             for (uint256 i = 0; i < numAgents; ++i) {
+                // Check for the zero bond value
+                if (agentParams[i].bond == 0) {
+                    revert ZeroValue();
+                }
```

Comment:
Bonds can never be zero as this way we would wrap zero values on the original ServiceRegistry side